com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainClassHandler.java

```diff
@@ -0,0 +1,131 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.microsoft.java.debug.plugin.internal;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.dom4j.Document;
+import org.dom4j.DocumentException;
+import org.dom4j.Element;
+import org.dom4j.io.SAXReader;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.search.IJavaSearchConstants;
+import org.eclipse.jdt.core.search.IJavaSearchScope;
+import org.eclipse.jdt.core.search.SearchEngine;
+import org.eclipse.jdt.core.search.SearchMatch;
+import org.eclipse.jdt.core.search.SearchParticipant;
+import org.eclipse.jdt.core.search.SearchPattern;
+import org.eclipse.jdt.core.search.SearchRequestor;
+
+import com.microsoft.java.debug.core.Configuration;
+
+public class ResolveMainClassHandler {
+    private static final Logger logger = Logger.getLogger(Configuration.LOGGER_NAME);
+
+    /**
+     * resolve main class and project name if not specified.
+     * @param arguments a list of arguments including project files and project name
+     * @return main class and project name
+     * @throws CoreException when there are error when resolving main class.
+     */
+    public String[] resolveMainClass(List<Object> arguments) throws CoreException {
+        List<String> projectNames;
+        if (arguments.size() > 1 && arguments.get(1) != null) {
+            // project name specified
+            projectNames = new ArrayList<>();
+            projectNames.add((String) arguments.get(1));
+        } else {
+            List<String> projectFiles = (List<String>) arguments.get(0);
+            projectNames = resolveProjectName(projectFiles);
+        }
+        String[] res = {null, null};
+        for (String projectName : projectNames) {
+            String mainClass = resolveMainClassCore(projectName);
+            if (mainClass != null) {
+                res[0] = mainClass;
+                res[1] = projectName;
+                return res;
+            }
+        }
+        return res;
+    }
+
+    private List<String> resolveProjectName(List<String> projectFiles) {
+        List<String> projects = new ArrayList<>();
+        for (String f : projectFiles) {
+            SAXReader reader = new SAXReader();
```

Comment:
Does eclipse have any built-in utility to help parse .project file?

Reply:
seems not